tests/integration/test_cluster.py

```diff
@@ -0,0 +1,240 @@
+        ("name__iexclude", ("VERY special CLUSTER", "Not a cluster")): num_clusters - 1,
+        ("name__contains", "special"): 1,
+        ("name__icontains", "-ClUsTeR-"): num_clusters - 2,
+        ("status__eq", "up"): 0,
+        ("status__eq", "down"): num_clusters,
+        ("status__in", ("down", "some status")): num_clusters,
+        ("status__in", ("up", "some status")): 0,
+        ("status__ne", "down"): 0,
+        ("status__ne", "up"): num_clusters,
+        ("status__exclude", ("excluded_status", "down")): 0,
+        ("status__exclude", ("excluded_status", "up")): num_clusters,
+        ("status__exclude", ("up", "down")): 0,
+    }
+    for filter_, expected in filters_data.items():
+        filter_value = {filter_[0]: filter_[1]}
+        clusters = await adcm_client.clusters.filter(**filter_value)
+        assert len(clusters) == expected, f"Filter: {filter_value}"
+
+
+async def _get_ansible_forks(adcm_client: ADCMClient, cluster: Cluster) -> int:
+    httpx_client, api_root = _get_httpx_client_base_url(adcm_client)
+    ansible_cfg_url = urljoin(api_root, f"clusters/{cluster.id}/ansible-config/")
+    response = await httpx_client.get(ansible_cfg_url)
+    assert response.status_code == 200
+
+    return response.json()["config"]["defaults"]["forks"]
+
+
+async def _update_cluster_name(adcm_client: ADCMClient, cluster: Cluster, new_name: str) -> None:
+    httpx_client, api_root = _get_httpx_client_base_url(adcm_client)
+    cluster_url = urljoin(api_root, f"clusters/{cluster.id}/")
+    response = await httpx_client.patch(cluster_url, json={"name": new_name})
+    assert response.status_code == 200
+
+
+async def _test_cluster_object_api(adcm_client: ADCMClient, cluster: Cluster, cluster_bundle: Bundle) -> None:
+    assert isinstance(cluster.id, int)
+    assert isinstance(cluster.name, str)
+    assert isinstance(cluster.description, str)
+
+    bundle = await cluster.bundle
+    assert isinstance(bundle, Bundle)
+    assert bundle.id == cluster_bundle.id
+
+    assert isinstance(await cluster.get_status(), str)
+    assert isinstance(await cluster.actions.all(), list)
+    assert isinstance(await cluster.upgrades.all(), list)
+    assert isinstance(await cluster.config_host_groups.all(), list)
+    assert isinstance(await cluster.action_host_groups.all(), list)
+    assert isinstance(await cluster.config, ObjectConfig)
+    assert isinstance(cluster.config_history, ConfigHistoryNode)
+    assert isinstance(await cluster.mapping, ClusterMapping)
+
+    initial_ansible_forks = await _get_ansible_forks(adcm_client, cluster)
+    await cluster.set_ansible_forks(value=initial_ansible_forks + 5)
+    assert await _get_ansible_forks(adcm_client, cluster) == initial_ansible_forks + 5
+
+    new_name = "New cluster name"
+    await _update_cluster_name(adcm_client, cluster, new_name)
+    assert cluster.name != new_name
+    await cluster.refresh()
+    assert cluster.name == new_name
+
+
+async def test_cluster_api(
```

Comment:
@a-alferov we need to discuss order and naming in tests

Reply:
refactored